mlflow-tests/tests/base.py

```
@@ -268,6 +268,34 @@ def cleanup_test_resources(self, admin_user_context):
+        # Cleanup MCP servers with workspace awareness
+        if self.test_context.mcp_servers_to_delete:
+            logger.info(f"Cleaning up {len(self.test_context.mcp_servers_to_delete)} MCP servers")
+            for server_name, workspace in self.test_context.mcp_servers_to_delete.items():
+                try:
+                    # Switch to the correct workspace
+                    if not self._switch_workspace(workspace, cleanup_errors):
+                        # Workspace switch failed, skip this resource
+                        continue
+
+                    try:
+                        server = mlflow.genai.get_mcp_server(server_name)
```

Ruff (0.16.1)

[error] Local variable `mlflow` referenced before assignment

(F823)